staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -354,47 +329,12 @@ func newETCD3Storage(c storagebackend.ConfigForResource, newFunc func() runtime.
 		// Hence, we only destroy once.
 		// TODO: fix duplicated storage destroy calls higher level
 		once.Do(func() {
-			stopCompactor()
-			stopDBSizeMonitor()
-			client.Close()
+			client.Done(tc.Transport)
```

Comment:
actually this is only called once https://github.com/kubernetes/kubernetes/pull/111559/files#r936279363

so I think that Done should clean immediately, independently of the number of refs

Reply:
I'm not following. The intent here is that multiple storage instances will end up sharing the same client. We don't want to close that client until no storage instances are using it anymore. Am I missing something?